lib/Dialect/Random/IR/RandomDialect.cpp

```diff
@@ -0,0 +1,81 @@
+#include "lib/Dialect/Random/IR/RandomDialect.h"
+
+#include <cassert>
+
+#include "llvm/include/llvm/ADT/TypeSwitch.h"            // from @llvm-project
+#include "llvm/include/llvm/Support/LogicalResult.h"     // from @llvm-project
+#include "mlir/include/mlir/IR/BuiltinTypes.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/DialectImplementation.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/TypeUtilities.h"          // from @llvm-project
+#include "mlir/include/mlir/Support/LLVM.h"              // from @llvm-project
+#include "mlir/include/mlir/Support/LogicalResult.h"     // from @llvm-project
+
+// NOLINTBEGIN(misc-include-cleaner): Required to define RandomDialect and
+// RandomOps
+#include "lib/Dialect/Random/IR/RandomOps.h"
+#include "lib/Dialect/Random/IR/RandomTypes.h"
+// NOLINTEND(misc-include-cleaner)
+
+// Generated definitions
+#include "lib/Dialect/Random/IR/RandomDialect.cpp.inc"
+
+#define GET_TYPEDEF_CLASSES
+#include "lib/Dialect/Random/IR/RandomTypes.cpp.inc"
+#define GET_OP_CLASSES
+#include "lib/Dialect/Random/IR/RandomOps.cpp.inc"
+
+namespace mlir {
+namespace heir {
+namespace random {
+
+void RandomDialect::initialize() {
+  addOperations<
+#define GET_OP_LIST
+#include "lib/Dialect/Random/IR/RandomOps.cpp.inc"
+      >();
+  addTypes<
+#define GET_TYPEDEF_LIST
+#include "lib/Dialect/Random/IR/RandomTypes.cpp.inc"
+      >();
+}
+
+LogicalResult DiscreteGaussianDistributionOp::verify() {
+  if (getStddev().getInt() < 0) {
+    return emitOpError()
+           << "Expected stddev to be non-negative, found stddev = "
+           << getStddev().getInt();
+  }
+  return success();
+}
```

Comment:
for this code, you could instead use this builtin AttrConstraint in the TD file: https://github.com/llvm/llvm-project/blob/f9e7cba122c2b636ddb975791aadf33c69f3f056/mlir/include/mlir/IR/CommonAttrConstraints.td#L768

Here's a use example: https://github.com/llvm/llvm-project/blob/708a9a06cba66bc8f739b05646e7d3be9247feee/mlir/test/lib/Dialect/Test/TestOps.td#L176